pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -221,9 +240,99 @@ func (fs *filesystem) newInode(creds *auth.Credentials, mode linux.FileMode) *ke
 
 // Open implements kernfs.Inode.Open.
 func (i *Inode) Open(ctx context.Context, rp *vfs.ResolvingPath, vfsd *vfs.Dentry, opts vfs.OpenOptions) (*vfs.FileDescription, error) {
-	fd, err := kernfs.NewGenericDirectoryFD(rp.Mount(), vfsd, &i.OrderedChildren, &i.locks, &opts)
-	if err != nil {
+	if opts.Flags&linux.O_LARGEFILE == 0 && i.size > linux.MAX_NON_LFS {
+		return nil, syserror.EOVERFLOW
+	}
+
+	fd := fileDescription{}
+	// Only send open request when the FUSE filesystem support open or is a directory.
+	if !i.fs.conn.NoOpen || opts.Mode.IsDir() {
+		kernelTask := kernel.TaskFromContext(ctx)
+		if kernelTask == nil {
+			log.Warningf("fusefs.DeviceFD.Read: couldn't get kernel task from context")
+			return nil, syserror.EINVAL
+		}
+
+		// Build the request.
+		var opcode linux.FUSEOpcode
+		if opts.Mode.IsDir() {
```

Comment:
What if the inode itself represents a directory? I think you also probably want to check
`i.InodeAttrs.Mode().IsDir()` here